sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/variantExpressions.scala

```diff
@@ -63,3 +70,300 @@ case class ParseJson(child: Expression) extends UnaryExpression
   override protected def withNewChildInternal(newChild: Expression): ParseJson =
     copy(child = newChild)
 }
+
+// A path segment in the `VariantGet` expression. It represents either an object key access (when
+// `key` is not null) or an array index access (when `key` is null).
```

Comment:
This doesn't like a good API design, shall we use scala `Either`?

Reply:
Good point, changed to `Either` instead.